packages/react-reconciler/src/ReactFiberHydrationContext.old.js

```diff
@@ -201,6 +203,7 @@ function warnNonhydratedInstance(returnFiber: Fiber, fiber: Fiber) {
       return;
     }
 
+    const isConcurrentMode = (returnFiber.mode & ConcurrentMode) !== NoMode;
```

Comment:
I tend to prefer to put these after the tag check. Especially since you don't need to read this field in all cases. Hoisting it out to encourage the anti-pattern of reading fields into variables but in a future refactor not all fields will be available on every type of tag.

Reply:
>the "follow the style of the surrounding code"-rule takes over.

👀

https://github.com/facebook/react/blob/5f7f5280836a44d4b81c1cb13f51e7482470cf1b/packages/react-reconciler/src/ReactFiberHydrationContext.old.js#L479